src/devices/sonoff.ts

```diff
@@ -545,6 +550,132 @@ const sonoffExtend = {
+            },
+        ];
+        const toZigbee: Tz.Converter[] = [
+            {
+                key: ['detach_relay_mode'],
+                convertSet: async (entity, key, value, meta) => {
+                    // logger.debug(`from zigbee 'key' ${key}`, NS);
+                    const detachRelay1 = 'detach_relay_outlet1';
+                    // logger.debug(`from zigbee detachRelay1: ${value[detachRelay1 as keyof typeof value]}`, NS);
+                    const detachRelay2 = 'detach_relay_outlet2';
+                    // logger.debug(`from zigbee detachRelay2: ${value[detachRelay2 as keyof typeof value]}`, NS);
+                    const detachRelay3 = 'detach_relay_outlet3';
+                    // logger.debug(`from zigbee detachRelay3: ${value[detachRelay3 as keyof typeof value]}`, NS);
+                    let detachRelayMask: number = 0;
+
+                    if (value[detachRelay1 as keyof typeof value] == 'ENABLE') {
+                        detachRelayMask |= 0x01;
+                    } else {
+                        detachRelayMask &= ~0x01;
+                    }
+                    if (value[detachRelay2 as keyof typeof value] == 'ENABLE') {
+                        detachRelayMask |= 0x02;
+                    } else {
+                        detachRelayMask &= ~0x02;
+                    }
+                    if (value[detachRelay3 as keyof typeof value] == 'ENABLE') {
+                        detachRelayMask |= 0x04;
+                    } else {
+                        detachRelayMask &= ~0x04;
+                    }
+                    // logger.info(`from zigbee detachRelayMask: ${detachRelayMask}`, NS);
+                    await entity.write(clusterName, {[attributeName]: detachRelayMask}, defaultResponseOptions);
+                    return {state: {[key]: value}};
+                },
+                convertGet: async (entity, key, meta) => {
+                    await entity.read(clusterName, [attributeName], defaultResponseOptions);
+                },
+            },
+        ];
+        return {
+            exposes: [exposes],
+            fromZigbee,
+            toZigbee,
+            isModernExtend: true,
+        };
+    },
+    sonoffOnOff(args: {powerOnBehavior?: boolean; skipDuplicateTransaction?: boolean; endpointNames?: string[]}): ModernExtend {
+        args = {powerOnBehavior: true, skipDuplicateTransaction: false, ...args};
+        const exposes: Expose[] = args.endpointNames ? args.endpointNames.map((ep) => e.switch().withEndpoint(ep)) : [e.switch()];
+        const fromZigbee: Fz.Converter[] = [args.skipDuplicateTransaction ? fz.on_off_skip_duplicate_transaction : fz.on_off];
+        const toZigbee: Tz.Converter[] = [tz.on_off];
+
+        if (args.powerOnBehavior) {
+            const behavior_string = ['off', 'on', 'toggle', 'previous'];
+            fromZigbee.push(fz.power_on_behavior);
+            toZigbee.push(tz.power_on_behavior);
+            if (args.endpointNames) {
```

Comment:
This can be moved into the `onOff` `modernExtend`, then `sonoffOnOff` can be removed.

Reply:
How should this be achieved? Since I tried to use onoff modernExtend before, only the first Endpoint is useful, while the others are not. Can you help me modify it?

Reply:
It has been added in the meantime, now a power on behaviour is exposed for each endpoint (https://github.com/Koenkk/zigbee-herdsman-converters/blob/46c7522dac972e5f71fd01967dc7c24a932ffcf2/src/lib/modernExtend.ts#L536).


You should be able to use it now.

Reply:
![M](https://github.com/user-attachments/assets/92b41fcc-d72c-4055-8ef5-4071acfdab63)
However, this does not solve my problem, I want each of the three channels to have different powerBehavior.